src/sentry/static/sentry/app/views/organizationIntegrationConfig.jsx

```diff
@@ -0,0 +1,259 @@
+`;
+
+function computeCenteredWindow(width, height) {
+  const screenLeft = window.screenLeft != undefined ? window.screenLeft : screen.left;
+  const screenTop = window.screenTop != undefined ? window.screenTop : screen.top;
+  const innerWidth = window.innerWidth
+    ? window.innerWidth
+    : document.documentElement.clientWidth
+      ? document.documentElement.clientWidth
+      : screen.width;
+
+  const innerHeight = window.innerHeight
+    ? window.innerHeight
+    : document.documentElement.clientHeight
+      ? document.documentElement.clientHeight
+      : screen.height;
+
+  const left = innerWidth / 2 - width / 2 + screenLeft;
+  const top = innerHeight / 2 - height / 2 + screenTop;
+
+  return {left, top};
+}
+
+export default class OrganizationIntegrationConfig extends AsyncView {
+  // TODO: proptypes
+
+  componentDidMount() {
+    this.dialog = null;
+    window.addEventListener('message', this.receiveMessage, false);
+  }
+
+  componentWillUnmount() {
+    super.componentWillUnmount();
+    window.removeEventListener('message', this.receiveMessage);
+
+    if (this.dialog !== null) {
+      this.dialog.close();
+    }
+  }
+
+  getTitle() {
+    if (this.state.config === null) {
+      return 'Global Integrations';
+    }
+
+    const {providerKey} = this.props.params;
+    const provider = this.state.config.providers.find(p => p.key == providerKey);
+
+    return `${provider.name} Integration`;
+  }
+
+  getEndpoints() {
+    const {orgId} = this.props.params;
+
+    return [
+      ['config', `/organizations/${orgId}/config/integrations/`],
+      ['itemList', `/organizations/${orgId}/integrations/`, {query: {status: ''}}],
+    ];
+  }
+
+  launchAddIntegration = provider => {
+    const name = 'sentryAddIntegration';
+
+    const {url, width, height} = provider.setupDialog;
+    const {left, top} = computeCenteredWindow(width, height);
+
+    this.dialog = window.open(
+      url,
+      name,
+      `scrollbars=yes,width=${width},height=${height},top=${top},left=${left}`
+    );
+
+    this.dialog.focus();
+    this.dialog.onclose = () => document.location.refresh();
+  };
+
+  receiveMessage = message => {
+    if (message.origin !== document.origin) {
+      return;
+    }
+
+    if (message.source !== this.dialog) {
+      return;
+    }
+
+    this.dialog = null;
+
+    const {success, data} = message.data;
+
+    if (!success) {
+      // TODO: Error state here
+      return;
+    }
+
+    let itemList = [...this.state.itemList, data];
+    itemList = sortArray(itemList, i => i.name);
+    itemList = _.uniqBy(itemList, i => i.id);
+
+    this.setState({itemList});
+  };
+
+  deleteIntegration = integration => {
+    const {orgId} = this.props.params;
+
+    const options = {
+      method: 'DELETE',
+      success: () =>
+        this.setState({
+          itemList: this.state.itemList.filter(item => item.id !== integration.id),
+        }),
+
+      // TODO toast indoactor for errors / complete
```

Comment:
Use `addSuccessMessage` and `addErrorMessage` action creators

Reply:
c08d1f0d85398d6cfc0f51cc9ee7890cc36d8485